browser/extensions/brave_extension_management.cc

```diff
@@ -37,4 +40,11 @@ void BraveExtensionManagement::RegisterForceInstalledExtensions() {
   UpdateForcedExtensions(&forced_list_pref);
 }
 
+void BraveExtensionManagement::RegisterBraveExtensions() {
+  const base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
+  if (!command_line.HasSwitch(switches::kDisableTorClientUpdaterExtension))
+    g_brave_browser_process->tor_client_updater()->Register();
```

Comment:
do we need this in BraveBrowserProcess? Seems like it could just live here?

Reply:
My thinking here was that `BraveBrowserProcess` would be the owner of long-lived services like the Tor client updater and then we'd just access it here via its getter so that we can register the extension on startup.

Are you suggesting to move the `Register` function into `BraveExtensionManagement` (or even the entire updater)? I haven't followed the code paths all the way through, but it seems like that could work. My only concerns would be that the registration happens at the correct time at startup (I definitely had a few issues with this before I settled on having `BraveBrowserProcess` own them) and that the updaters are easily accessible to other parts of the code.

Reply:
the tor client updater isn't a long-lived service, the component updater is the long-lived service and you are just registering things to be updated by it

Reply:
I think maybe there is some confusion here because of the naming. I would use the WidevineCDMComponentInstaller as an example for this

Reply:
OK, I see what you're saying about component updater. I did model this on some of the existing components (not specifically `WidevineCDMComponentInstaller`, but something similar). In our code, `BraveComponentInstaller` is fulfilling the same role as `WidevineCDMComponentInstaller` (i.e., it implements the installer policy and provides a register function).

It seems like making `BraveComponentInstaller` generic was a mistake; instead, it should be specific to this component and contain all of the logic that gets executed upon registration. Also, it seems like registration should occur in `chrome_browser_main` or somewhere similar. Is that the approach you're thinking of?

Reply:
probably in BraveBrowserMainExtraParts::PreMainMessageLoopRun

Reply:
btw we'll want this opt in and not always registered.  Installed first time switch is turned on in a private tab. 
@cezaraugusto can you track this work and do the fix for the private tab UI? You could land a dead switch control any time.